sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -219,13 +219,21 @@ object JdbcUtils extends Logging {
     val sb = new StringBuilder()
     val dialect = JdbcDialects.get(url)
     df.schema.fields foreach { field => {
-      val name = field.name
+      val name = dialect.quoteIdentifier(field.name)
       val typ: String = getJdbcType(field.dataType, dialect).databaseTypeDefinition
       val nullable = if (field.nullable) "" else "NOT NULL"
       sb.append(s", $name $typ $nullable")
     }}
     if (sb.length < 2) "" else sb.substring(2)
   }
+  
+  /**
+   * Parse the table name string for this RDD.
```

Comment:
This is confusing, since the variable `table` already sounds like it would be the table name.

Can you call it `schemaQualifiedTableName` or something more explicit?